homeassistant/components/daikin/__init__.py

```diff
@@ -147,15 +148,32 @@ def _update_unique_id(entity_entry: er.RegistryEntry) -> dict[str, str] | None:
     if new_unique_id == old_unique_id:
         return
 
+    duplicate = dev_reg.async_get_device(
+        connections={(CONNECTION_NETWORK_MAC, new_mac)}, identifiers=None
+    )
+
+    # Remove duplicated device
+    if duplicate is not None:
+        _LOGGER.debug(
+            "Removing duplicated device %s",
+            duplicate.name,
+        )
+
+        duplicate_entities = er.async_entries_for_device(ent_reg, duplicate.id, True)
+        for entity in duplicate_entities:
+            ent_reg.async_update_entity(
+                entity.entity_id, new_unique_id=f"old-{entity.unique_id}"
+            )
+
+        dev_reg.async_remove_device(duplicate.id)
+
```

Comment:
What's happening here?
What created the duplicated device? Do we need to do another check to make sure we're removing the correct device?

Also, `dev_reg.async_remove_device(duplicate.id)` will remove any entities owned by the removed device, so why do we first iterate over the device's entities changing theri unique_ids?

Reply:
The duplicates were created by the pydaikin library having a bug. The duplicates use the new mac address which I check for. I renamed the unique ids to stop a race condition with migrating the existing entities.

Reply:
> I renamed the unique ids to stop a race condition with migrating the existing entities

Can you elaborate on that please, where is the race?

Reply:
> The duplicates use the new mac address which I check for.

OK, so if the config entry's unique ID is not yet migrated but there's already a device with the new mac address that's a duplicate?

Reply:
It's not really a race but the entities aren't removed before the unique id migration runs causing the unique id in use error.

Reply:
> > The duplicates use the new mac address which I check for.
> 
> OK, so if the config entry's unique ID is not yet migrated but there's already a device with the new mac address that's a duplicate? Shouldn't you also check if the device you find is connected to the right config entry?

Yes and I'll add a check.

Reply:
Yes that makes sense, i'll make the change